pyfr/solvers/euler/elements.py

```diff
@@ -49,6 +50,78 @@ def con_to_pri(cons, cfg):
 
         return [rho] + vs + [p]
 
+    @staticmethod
+    def validate_formulation(form, intg, cfg):
+        if form not in BaseFluidElements.formulations:
+            raise ValueError('System not compatible with time stepping formulation.')
+
+        shock_capturing = cfg.get('solver', 'shock-capturing')
```

Comment:
Wrap in a check to ensure that the solver/shock-capturing key exists first (as I think it will fail otherwise).